Base.rte/Effects/Pyro/Flame/GroundFlameNew.lua

```diff
@@ -0,0 +1,66 @@
+function Create(self)
+
+	self.strength = 80;	-- damage output
+	
+	self.checkTimer = Timer();
+	self.checkDelay = math.random(20);
+end
+
+function Update(self)
+
+	self.ToSettle = false;
+
+	if self.target and IsMOSRotating(self.target) then
+
+		self.Vel = self.target.Vel;
+		self.Pos = self.target.Pos - Vector(self.stickPos.X, self.stickPos.Y):RadRotate(self.target.RotAngle - self.tsAngle);
+
+		local actor = MovableMan:GetMOFromID(self.target.RootID);	-- inflict damage on the root actor
+		if MovableMan:IsActor(actor) then
+			actor = ToActor(actor);
+			if math.random() < (self.strength * self.target.DamageMultiplier) / actor.Mass / (self.target.Material.StructuralIntegrity) then
+				actor.Health = actor.Health - 1;
+			end
+		end
+	else
+		self.target = nil;
+		if self.checkTimer:IsPastSimMS(self.checkDelay) then
+		
+			self.checkTimer:Reset();
+			self.checkDelay = self.checkDelay + 1;	-- Extend the delay for optimization reasons
+
+			local checkPos = self.Pos + Vector(self.Vel.X, self.Vel.Y):SetMagnitude(math.sqrt(self.Vel.Magnitude));
+			local mocheck = SceneMan:GetMOIDPixel(checkPos.X, checkPos.Y);
+			if mocheck ~= 255 then
+				local mo = MovableMan:GetMOFromID(mocheck);
+				if mo then
+					self.target = ToMOSRotating(mo);
+
+					self.tsAngle = mo.RotAngle;	-- target angle when sticking	
+
+					self.stickPos = SceneMan:ShortestDistance(mo.Pos, self.Pos, SceneMan.SceneWrapsX) * 0.8;
+
+					self.Pos = mo.Pos - Vector(self.stickPos.X, self.stickPos.Y);
+				end
+			end
+		end
+	end
+
+	local age = self.Age * 0.0001 + 1;
+	local chance = math.random();
+	local part;
```

Comment:
How about `particle` or something? part could be short for a whole bunch of things
Also, whitespace on line 56 before the elseif.